server/src-lib/Hasura/SQL/Value.hs

```diff
@@ -127,14 +127,35 @@ parsePGValue ty val = case (ty, val) of
   (_          , String t) -> parseTyped <|> pure (PGValUnknown t)
   (_          , _)        -> parseTyped
   where
+    parseBoundedInt :: forall i. (Integral i, Bounded i) => Value -> AT.Parser i
+    parseBoundedInt val' = do
+      parsedM <-
+        withScientific
+          ("Integer expected for input type: " ++ show ty)
+          (pure . toBoundedInteger)
+          val'
+      case parsedM of
+        Just parsed -> return parsed
+        Nothing     -> fail $ "Integer does not fit.  Maybe it is a float,"
```

Comment:
It would be more helpful to have a message which includes the provided value and the type, maybe something like this?
```
the value '....' exceeds the bounds of type 'int'
```

Reply:
or `lies outside the bounds`